modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -1643,10 +1643,11 @@ public Source getEventSource(Event event) {
 
   @Override
   public Opt<WorkflowInstance> getCurrentWorkflowInstance(String mpId) throws IndexServiceException {
-    WorkflowQuery query = new WorkflowQuery().withMediaPackage(mpId);
+//    WorkflowQuery query = new WorkflowQuery().withMediaPackage(mpId);
```

Comment:
```
Keep the generic jobs and use oc_job_argument to store the media package 
identifier instead...
```
I had initially wanted to go down this route. Saving the media package identifier would be easy enough as an argument could pretty much be anything we'd like it to be.
But it left me with these questions:

- the `argument` column in `oc_job_argument` is of type `longtext`, and I'm not too clued up on how this affects indexing (I mean, we'd probably want to add an index on the mediapackage ids right?),
- suppose we query a (Jpa)Job, lets call it `job`. One of the entries in the list `job.arguments` would be the media package identifier. As a first approach, I'd probably use a regex to determine the `mpId`. But because we can add as many arguments as we'd like (those could be additional `mpId`s as arguments), we need further checks to assure that the mapping between workflows and media packages is well-defined. Ultimately, I'd prefer an approach that assures a well-defined mapping than having to code those checks.

````
Separate jobs and workflows...
```
It definitely would be more work, but imo _that_ shouldn't be a determining factor... code tends to stick around so do it right the first time round.

If you don't mind, I'll get you on IRC this week for a chat on best strategies for this patch.